src/object.c

```diff
@@ -1582,32 +1575,39 @@ sds getMemoryDoctorReport(void) {
     return s;
 }
 
+/* Return the LFU frequency for an object. */
+uint8_t objectGetLFUFrequency(robj *o) {
+    uint8_t freq;
+    o->lru = lfu_getFrequency(o->lru, &freq);
+    return freq;
+}
+
+/* Return the LRU idle time for an object. */
+uint32_t objectGetLRUIdleSecs(robj *o) {
+    return lru_getIdleSecs(o->lru);
+}
+
+/* Return an indication of idleness.  Larger numbers are more idle. */
+uint32_t objectGetIdleness(robj *o) {
+    uint32_t idleness;
+    o->lru = lrulfu_getIdleness(o->lru, &idleness);
+    return idleness;
+}
+
 /* Set the object LRU/LFU depending on server.maxmemory_policy.
  * The lfu_freq arg is only relevant if policy is MAXMEMORY_FLAG_LFU.
  * The lru_idle and lru_clock args are only relevant if policy
  * is MAXMEMORY_FLAG_LRU.
  * Either or both of them may be <0, in that case, nothing is set. */
-int objectSetLRUOrLFU(robj *val, long long lfu_freq, long long lru_idle, long long lru_clock, int lru_multiplier) {
-    if (server.maxmemory_policy & MAXMEMORY_FLAG_LFU) {
+int objectSetLRUOrLFU(robj *val, long long lfu_freq, long long lru_idle_secs) {
+    if (lrulfu_isUsingLFU()) {
         if (lfu_freq >= 0) {
             serverAssert(lfu_freq <= 255);
```

Comment:
if we are refactoring...
```suggestion
            serverAssert(lfu_freq <= UINT8_MAX);
```